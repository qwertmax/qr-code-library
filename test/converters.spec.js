import {
  convertToUrl,
  convertToText,
  convertToVCard,
  convertToEmail,
  convertToSms
} from '../src/converters';

test('convertToUrl should return passed string', () => {
  expect(convertToUrl('http://url.com')).toEqual('http://url.com');
});

test('convertToText should return passed string', () => {
  expect(convertToText('text')).toEqual('text');
});

describe('convertToVCard', () => {
  it('should return vCard representation for last name and first name', () => {
    const payload = {
      firstName: 'John',
      lastName: 'Doe'
    };

    expect(convertToVCard(payload)).toMatchSnapshot();
  });

  it('should return vCard representation with additional fields', () => {
    const payload = {
      firstName: 'John',
      lastName: 'Doe',
      organization: 'Company',
      role: 'Developer',
      email: 'test@example.com',
      url: 'test.example.com'
    };

    expect(convertToVCard(payload)).toMatchSnapshot();
  });

  it('should return vCard representation with address', () => {
    const payload = {
      firstName: 'John',
      lastName: 'Doe',
      street: '123 Main Street',
      city: 'Any Town',
      state: 'CA',
      zip: '91921-1234',
      country: 'USA'
    };

    expect(convertToVCard(payload)).toMatchSnapshot();
  });
});

describe('convertToEmail', () => {
  it('should return email uri', () => {
    expect(convertToEmail({ email: 'test@example.com' })).toEqual(
      'mailto:test@example.com'
    );
  });
  it('should return email uri with additional queries', () => {
    const payload = {
      email: 'test@example.com',
      subject: 'Look here! Email subject!',
      body: 'Something...',
      cc: 'cc@example.com',
      bcc: 'bcc@example.com'
    };

    expect(convertToEmail(payload)).toMatchSnapshot();
  });
});

test('convertToSms should return sns uri', () => {
  expect(convertToSms({ tel: '+71234567890', message: 'Sms text' })).toEqual(
    'smsto:+71234567890:Sms text'
  );
});
