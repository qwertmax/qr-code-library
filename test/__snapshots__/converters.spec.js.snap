// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertToEmail should return email uri with additional queries 1`] = `"mailto:test@example.com?&subject=Look here! Email subject!&body=Something...&cc=cc@example.com&bcc=bcc@example.com"`;

exports[`convertToVCard should return vCard representation for last name and first name 1`] = `
"BEGIN:VCARD
N:Doe;John
FN: John Doe
ADR;TYPE=WORK:;;;;;;
END:VCARD
"
`;

exports[`convertToVCard should return vCard representation with additional fields 1`] = `
"BEGIN:VCARD
N:Doe;John
FN: John Doe
ORG:Company
TITLE: Developer
TEL;TYPE=work,voice:+5550000
ADR;TYPE=WORK:;;;;;;
EMAIL:test@example.com
URL:test.example.com
END:VCARD
"
`;

exports[`convertToVCard should return vCard representation with address 1`] = `
"BEGIN:VCARD
N:Doe;John
FN: John Doe
ADR;TYPE=WORK:;;123 Main Street;Any Town;CA;91921-1234;USA
END:VCARD
"
`;
